<?xml version="1.0"?>
<launch>
<!-- arguments -->
<arg name="domain_path" default="$(find exp_assignment3)/knowledge/cluedo_dom.pddl" />
<arg name="problem_path" default="$(find exp_assignment3)/knowledge/cluedo_prob.pddl" />
<!-- knowledge base -->
<node name="rosplan_knowledge_base" pkg="rosplan_knowledge_base" type="knowledgeBase" respawn="false" output="screen">
<param name="domain_path" value="$(arg domain_path)" />
<param name="problem_path" value="$(arg problem_path)" />
<!-- conditional planning flags -->
<param name="use_unknowns" value="false" />
</node>
<!-- problem generation -->
<include file="$(find rosplan_planning_system)/launch/includes/problem_interface.launch">
<arg name="knowledge_base" value="rosplan_knowledge_base" />
<arg name="domain_path" value="$(arg domain_path)" />
<arg name="problem_path" value="$(find rosplan_demos)/common/new_problem.pddl" />
<arg name="problem_topic" value="problem_instance" />
</include>

<include file="$(find rosplan_planning_system)/launch/includes/planner_interface.launch">
<arg name="use_problem_topic" value="true" />
<arg name="problem_topic" value="/rosplan_problem_interface/problem_instance" />
<arg name="planner_topic" value="planner_output" />
<arg name="domain_path" value="$(arg domain_path)" />
<arg name="data_path" value="$(find rosplan_demos)/common/" />
<arg name="planner_command" value="timeout 10 $(find rosplan_planning_system)/common/bin/popf DOMAIN PROBLEM" />
</include>

<node name="rosplan_parsing_interface" pkg="rosplan_planning_system" type="pddl_simple_plan_parser" respawn="false" output="screen">
<param name="knowledge_base" value="rosplan_knowledge_base" />
<param name="planner_topic" value="/rosplan_planner_interface/planner_output" />
<param name="plan_topic" value="complete_plan" />
</node>

<node name="rosplan_plan_dispatcher" pkg="rosplan_planning_system" type="pddl_simple_plan_dispatcher" respawn="false" output="screen">
<param name="knowledge_base" value="rosplan_knowledge_base" />
<param name="plan_topic" value="/rosplan_parsing_interface/complete_plan" />
<param name="action_dispatch_topic" value="action_dispatch" />
<param name="action_feedback_topic" value="action_feedback" />
</node>

<!-- sim actions -->
        
	<!--include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="goto_waypointt" />
		<arg name="action_duration" value="1" />
		
	</include-->
	<node name="rosplan_interface_go_to_waypointt" pkg="exp_assignment3" type="go_to_waypointt" respawn="false" output="screen">
	        <param name="knowledge_base" value="rosplan_knowledge_base" />
	        <param name="pddl_action_name" value="goto_waypointt" />
	        <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
	        <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>
	
	
	<!--include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="get_two_hint" />
		<arg name="action_duration" value="1" />
		<arg name="action_probability" value="1.0"/>
	</include-->
	<node name="rosplan_interface_get_two_hint" pkg="exp_assignment3" type="get_two_hint" respawn="false" output="screen">
	        <param name="knowledge_base" value="rosplan_knowledge_base" />
	        <param name="pddl_action_name" value="get_two_hint" />
	        <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
	        <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>
	
	<!--include file="$(find rosplan_planning_system)/launch/includes/simulated_action.launch" >
		<arg name="pddl_action_name" value="check_hypothesis_correctt" />
		<arg name="action_duration" value="1" />
		<arg name="action_probability" value="1.0"/>
	</include-->
	<node name="rosplan_interface_check_hyp_correctt" pkg="exp_assignment3" type="check_hyp_correctt" respawn="false" output="screen">
	        <param name="knowledge_base" value="rosplan_knowledge_base" />
	        <param name="pddl_action_name" value="check_hypothesis_correctt" />
	        <param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
	        <param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>
</launch>
